homeassistant/components/bkk/sensor.py

```diff
@@ -0,0 +1,314 @@
+
+    def __init__(self, stop_id, idx, vehicle_type):
+        """Initialize the sensor."""
+        self._id = self._line = ""
+        self._name = None
+        self._entity_idx = idx
+        self._stop_id = stop_id
+        self._departure_time = 0
+        self._predicted_departure_time = 0
+        self._route_id = ""
+        if vehicle_type in ICONS:
+            self._icon = ICONS[vehicle_type]
+        else:
+            self._icon = "mdi:bus-alert"
+        self._attributes = {
+            "vehicle": vehicle_type,
+            "line": self._line,
+            "route_id": self._route_id,
+        }
+        self._state = None
+
+    def clear_fields(self):
+        """Clear BKKRoute entity fields."""
+        self._name = "N/A"
+        self._departure_time = 0
+        self._predicted_departure_time = 0
+        self._attributes["wheelchair_accessible"] = ""
+        self._attributes["destination"] = ""
+        self._state = None
+
+    def update_fields(self, current_stop_time, routes, trips):
+        """Update BKKRouter entity fields with fresh data."""
+        self._departure_time = (
+            current_stop_time["departureTime"]
+            if "departureTime" in current_stop_time
+            else 0
+        )
+        self._predicted_departure_time = (
+            current_stop_time["predictedDepartureTime"]
+            if "predictedDepartureTime" in current_stop_time
+            else 0
+        )
+        self._line = routes[trips[current_stop_time["tripId"]]["routeId"]]["shortName"]
+        self._id = routes[trips[current_stop_time["tripId"]]["routeId"]]["id"]
+        self._name = (
+            f"{self.line}->{trips[current_stop_time['tripId']]['tripHeadsign']}"
+        )
+        self._attributes["line"] = self._line
+        self._attributes["route_id"] = self._id
+        self._attributes["wheelchair_accessible"] = trips[current_stop_time["tripId"]][
+            "wheelchairAccessible"
+        ]
+        self._attributes["destination"] = trips[current_stop_time["tripId"]][
+            "tripHeadsign"
+        ]
+        self.calculate_state()
+
+    @property
+    def icon(self):
+        """Return entity icon."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return route state."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return state attributes."""
+        return self._attributes
+
+    @property
+    def name(self):
+        """Return entity name."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return measurement unit."""
+        return "min"
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return f"{self._stop_id}_{self._entity_idx}"
+
+    def update(self):
+        """Update entity state."""
+        pass
+
+    @staticmethod
+    def to_iso_datetime(epoch):
+        """Convert epoch date format to ISO format."""
+        if epoch > 0:
```

Comment:
Use it as a truthy value:
```suggestion
        if epoch:
```